internal/testutils/networking/l2_segment_linux.go

```diff
@@ -0,0 +1,339 @@
+package networking
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+	"runtime"
+	"syscall"
+	"testing"
+
+	"github.com/docker/docker/libnetwork/testutils"
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+	"gotest.tools/v3/assert"
+)
+
+// Netns represents a network namespace that might be named, and thus mounted under /run/netns.
+type Netns struct {
+	handle netns.NsHandle
+	name   string
+}
+
+// CurrentNetns returns an unnamed Netns for the current network namespace this thread is living in. If it fails to
+// get a handle for the current net namespace, it bails out.
+func CurrentNetns(t *testing.T) Netns {
+	t.Helper()
+
+	curNs, err := netns.Get()
+	assert.NilError(t, err)
+
+	return Netns{handle: curNs}
+}
+
+// NewNamedNetns creates a new named Netns and then switch back to the original net namespace. It bails out if it fails.
+func NewNamedNetns(t *testing.T, name string, reusePrevious bool) Netns {
+	t.Helper()
+
+	var err error
+	newNs := Netns{name: name}
+	newNs.handle, err = netns.GetFromName(newNs.name)
+	if err != nil && !os.IsNotExist(err) {
+		assert.NilError(t, err)
+	}
```

Comment:
Does this assert actually do anything? It would always be a non-nil error? Should this be a `t.Fatal(err)`?